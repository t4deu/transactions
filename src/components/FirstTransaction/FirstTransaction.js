// @flow
// external imports
import React from 'react';
import { View, Text } from 'react-native';
import { Card, Button } from 'react-native-material-ui';
// styles
import styles from './FirstTransaction.styles';

type Props = {
  onAddCredit: Function,
};

const FirstTransaction = (props: Props) => (
  <View style={styles.painel}>
    <Card>
      <View style={styles.card}>
        <Text style={styles.welcome}>Seja Bem Vindo!</Text>
        <Text style={styles.text}>
            Toque neste botão aqui para adicionar nossa primeira transação
        </Text>
        <Button onPress={props.onAddCredit} raised primary icon="add" text="Adicionar Renda" />
      </View>
    </Card>
  </View>
);

export default FirstTransaction;
