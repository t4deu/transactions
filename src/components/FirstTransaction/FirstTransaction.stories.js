import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { View } from 'react-native';

import FirstTransactions from './FirstTransaction';

storiesOf('FirstTransactions', module)
  .addDecorator(story => <View style={{ flex: 1, marginTop: 20 }}>{story()}</View>)
  .add('Credit', () => (
    <FirstTransactions onAddCredit={() => {}} />
  ));
