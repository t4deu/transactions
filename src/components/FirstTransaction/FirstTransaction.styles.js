export default {
  painel: {
    flex: 1,
    paddingTop: 80,
    paddingBottom: 20,
    backgroundColor: '#fafafa',
  },
  card: {
    padding: 20,
    paddingBottom: 30,
  },
  welcome: {
    fontSize: 28,
    textAlign: 'center',
    marginBottom: 50,
  },
  text: {
    fontSize: 16,
    textAlign: 'center',
    marginBottom: 50,
  },
  currency: {
    color: '#8600c9',
    fontWeight: 'bold',
  },
  total: {
    color: '#8600c9',
    fontWeight: 'bold',
    fontSize: 18,
    textAlign: 'center',
    marginVertical: 5,
  },
  advice: {
    fontStyle: 'italic',
  },
};
