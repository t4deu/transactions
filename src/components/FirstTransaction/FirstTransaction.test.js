import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import FirstTransaction from './FirstTransaction';

describe('<FirstTransaction>', () => {
  it('renders without issues', () => {
    const component = shallow(<FirstTransaction onAddCredit={() => {}} />);

    expect(toJson(component)).toMatchSnapshot();
  });
});
