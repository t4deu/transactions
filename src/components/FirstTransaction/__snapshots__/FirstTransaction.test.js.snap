// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FirstTransaction> renders without issues 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fafafa",
      "flex": 1,
      "height": 200,
      "marginTop": 60,
      "paddingVertical": 20,
    }
  }
>
  <Card
    onPress={null}
    style={Object {}}
  >
    <View
      style={
        Object {
          "padding": 20,
          "paddingBottom": 30,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 28,
            "marginBottom": 50,
            "textAlign": "center",
          }
        }
      >
        Seja Bem Vindo!
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 16,
            "marginBottom": 50,
            "textAlign": "center",
          }
        }
      >
        Toque neste botão aqui para adicionar nossa primeira transação
      </Text>
      <Button
        accent={false}
        disabled={false}
        icon="add"
        onLongPress={null}
        onPress={[Function]}
        primary={true}
        raised={true}
        style={Object {}}
        text="Adicionar Renda"
        upperCase={true}
      />
    </View>
  </Card>
</View>
`;
